public/js/src/game/ui/HanabiCardClick.js

```diff
@@ -58,14 +58,24 @@ const clickLeft = (card, event) => {
 };
 
 const clickMiddle = (card, event) => {
-    // No actions in this function use modifiers
-    if (event.ctrlKey || event.shiftKey || event.altKey || event.metaKey) {
+    // No actions in this function use modifiers other than alt
+    if (event.ctrlKey || event.shiftKey || event.metaKey) {
         return;
     }
 
-    // Middle clicking on cards goes to the turn it was first clued
-    if (card.turnClued !== null) {
-        goToTurn(card.turnClued);
+    // Middle clicking on cards goes to a turn it was clued
+    if (card.turnsClued.length !== 0 && card.turnsClued[0] !== globals.turn) {
+        // Alt-middle clicking goes to turn it was first clued
+        if (event.altKey) {
+            goToTurn(card.turnsClued[0]);
+            // No modifier goes to turn it was last clued
+        } else {
+            let turnToGoTo = card.turnsClued.pop();
```

Comment:
Talking about the order of events in this code is sometimes interesting because of the rewind mechanism. What interactions did you have in mind? This code does not make that clear.

Do you mean the latest turn before the current position?

Reply:
Yes, so it doesn't go forwards in time or get stuck on current turn.

Reply:
Without the while-loop, it rebuilt itself every time, require more and more middle-clicks to pop off future turns.

Reply:
@Zamiell did you make `HanabiCard` be preserved by rewinds? Otherwise I have no idea what to make of Asddsa's statement.

Reply:
And if that did happen, it strongly suggests a change to your approach, which would be constantly building and tearing down the array.

Reply:
Seems like an off-by-one.